src/Device.Net/ControlTransferResult.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Device.Net
```

Comment:
@josesimoes why is this necessary?

It's exactly the same as ReadResult

Reply:
Almost... the property BytesTransferred name is different to make it clear that's the bytes transferred. 
That is properly set for both send and receive operations.

Reply:
Hmm...

It's still reading the bytes from the device... Is the distinction that important?

Reply:
@josesimoes in my branch I fixed the code analyzer rule.

csharp_style_expression_bodied_operators = true:error

Reply:
I believe it is. Because when you are sending data it will report how many bytes where actually SENT.
And you're reading data it will tell you how many bytes where actually RECEIVED.

On both that's the gold check for the operation actually succeeding. Not only the return from the call.

Plus on the name, I feel that ReadResult is a bit... short... because it's being used for READ *AND* WRITE operations, therefore the name ControlTransferResult which works for both.

Reply:
I could change the name to TransferResult, and the property to BytesTransferred but this would break all the existing implementations.

Reply:
ReadResult is not being used for write operations. It's only used for reading from the device.

Reply:
@josesimoes after reading this a few times, I'm more confused now.

When we do a control transfer, we send a buffer of data to the device. Then, the device sends the buffer back. So, technically, we may be interested in the transfer length for both operations (send and receive). However, the Windows API does not give us that. We only get the length of the data that was read back from the device. Isn't that understanding correct?

Reply:
In all cases, what we get is a Read Result.

Reply:
You're the boss... that's your call...
ReadResult is not used for write operations because there is none, yet. (I guess).
Where there is the name will be kind of awkward, no?!

Suggestion: 
1) keep the existing ReadResult and methods and add the [Obsolete] attribute to them.
2) Implement the "new" TransferResult with those calls.
3) At some future version remove the deprecated ones.

Reply:
> @josesimoes after reading this a few times, I'm more confused now.
> 
> When we do a control transfer, we send a buffer of data to the device. Then, the device sends a buffer back with different data. So, technically, we may be interested in the transfer length for both operations (send and receive). However, the Windows API does not give us that. We only get the length of the data that was read back from the device. Isn't that understanding correct?

Nope... The LengthTransferred in WinUsb_ControlTransfer reports the actuall number of bytes that were SENT or RECEIVED according to the operation that was requested (RequestDirection.In or RequestDirection.Out).

Reply:
No @josesimoes. you're pointing out something that I must be misunderstanding. I need to get this straight.

You're saying that the you want to use the structure for the result of sends, and receives, but the Windows API doesn't give us the result of sends - only receive. We only get the read result of what comes back from the device., 

This is the method signature:

```cs
public static extern bool WinUsb_ControlTransfer(IntPtr InterfaceHandle, WINUSB_SETUP_PACKET SetupPacket, byte[] Buffer, uint BufferLength, ref uint LengthTransferred, IntPtr Overlapped);
```

I could be wrong, but my understanding is that `LengthTransferred` is the length read from the device.

Reply:
Because the operation does both a send and a receive, it's almost like we need two results.

Reply:
Please check here: https://docs.microsoft.com/en-us/windows/win32/api/winusb/nf-winusb-winusb_controltransfer


Depending if you're sending data to the device or reading from it, it will tell you how many bytes were actually transferred. I insist! 😉

Reply:
> Nope... The LengthTransferred in WinUsb_ControlTransfer reports the actuall number of bytes that were SENT or RECEIVED according to the operation that was requested (RequestDirection.In or RequestDirection.Out).

OK. That's where my misunderstanding was. Thanks